org-code-javabuilder/playground/src/main/java/org/code/playground/TextItem.java

```diff
@@ -85,19 +83,58 @@ public String getText() {
    * @param color the text color
    */
   public void setColor(Color color) {
-    this.setColorHelper(color);
+    // Copy to new color to store the values only, and not a reference to the original color object
+    this.color = new Color(color);
     HashMap<String, String> colorDetails = new HashMap<>();
     this.addColorToDetails(colorDetails);
     this.sendChangeMessage(colorDetails);
   }
 
+  /**
+   * Set the amount of red (ranging from 0 to 255). Values below 0 will be ignored and set to 0, and
+   * values above 255 will be ignored and set to 255.
+   *
+   * @param colorRed the amount of red (ranging from 0 to 255) in the color of the text.
+   */
+  public void setRed(int colorRed) {
+    this.color = new Color(colorRed, this.color.getGreen(), this.color.getBlue());
+    this.sendChangeMessage(
+        ClientMessageDetailKeys.COLOR_RED, Integer.toString(this.color.getRed()));
+  }
+
+  /**
+   * Set the amount of green (ranging from 0 to 255). Values below 0 will be ignored and set to 0,
+   * and values above 255 will be ignored and set to 255.
+   *
+   * @param colorGreen the amount of green (ranging from 0 to 255) in the color of the text.
+   */
+  public void setGreen(int colorGreen) {
+    this.color = new Color(this.color.getRed(), colorGreen, this.color.getBlue());
+    this.sendChangeMessage(
+        ClientMessageDetailKeys.COLOR_GREEN, Integer.toString(this.color.getGreen()));
+  }
+
+  /**
+   * Set the amount of blue (ranging from 0 to 255). Values below 0 will be ignored and set to 0,
+   * and values above 255 will be ignored and set to 255.
+   *
+   * @param colorBlue the amount of blue (ranging from 0 to 255) in the color of the text.
+   */
+  public void setBlue(int colorBlue) {
+    this.color = new Color(this.color.getRed(), this.color.getGreen(), colorBlue);
+    this.sendChangeMessage(
+        ClientMessageDetailKeys.COLOR_BLUE, Integer.toString(this.color.getBlue()));
+  }
+
   /**
    * Get the text color for the item.
    *
    * @return the text color for the item
    */
   public Color getColor() {
-    return this.color;
+    // TODO: wrapping this.color in a new Color object will not be necessary once Color is
+    // immutable.
```

Comment:
NIT: Grab a permalink to this TODO and add it to the task to make color immutable. (And similarly to other TODOs in related to making color immutable in this task). The task, for reference: https://codedotorg.atlassian.net/browse/CSA-865

Reply:
Oh yep sure - do we usually link to the PR itself or the permalink once it's merged?

Reply:
I generally make a GitHub comment and link to the comment. Permalink would work too, but it's a lot of steps to remember.